core/templates/components/summary-tile/lesson-card.component.ts

```diff
@@ -0,0 +1,153 @@
+  topicFragment: string | undefined;
+  story: string;
+  currentStory: StoryNode;
+}
+
+interface ThumbnailUrl {
+  filename: string;
+  id: string;
+}
+@Component({
+  selector: 'lesson-card',
+  templateUrl: './lesson-card.component.html',
+})
+export class LessonCardComponent implements OnInit {
+  @Input() story!: StorySummary | LearnerExplorationSummary | CollectionSummary;
+  @Input() topic!: string;
+
+  desc!: string;
+  imgColor!: string;
+  imgUrl!: string;
+  lessonUrl!: string;
+  progress!: number;
+  title!: string;
+  lessonTopic!: string;
+
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private assetsBackendApiService: AssetsBackendApiService
+  ) {}
+
+  ngOnInit(): void {
+    if (this.story instanceof StorySummary) {
+      let completedStories = this.story.getCompletedNodeTitles().length;
+
+      this.desc = this.story.getTitle();
+      this.imgColor = this.story.getThumbnailBgColor();
+      this.imgUrl = this.getStorySummaryThumbnailUrl({
+        filename: this.story.getThumbnailFilename(),
+        id: this.story.getId(),
+      });
+
+      let nextStory =
+        completedStories === this.story.getAllNodes().length
+          ? completedStories - 1
+          : completedStories;
+      /* Last completed story index works because if 1 is completed, 1 index is 2nd item */
+      let lessonArgs = {
+        classroom: this.story.getClassroomUrlFragment(),
+        topicFragment: this.story.getTopicUrlFragment(),
+        currentStory: this.story.getAllNodes()[nextStory],
+        story: this.story.getUrlFragment(),
+      };
+      this.lessonUrl = this.getStorySummaryLessonUrl(lessonArgs);
+
+      this.title = `Chapter ${nextStory + 1}: ${this.story.getNodeTitles()[nextStory]}`;
+      this.progress = Math.floor(
+        (completedStories / this.story.getNodeTitles().length) * 100
+      );
+      this.lessonTopic = this.topic;
+    } else {
+      /* Implementation for explorations & collections */
+      this.desc = this.story.objective;
+      this.imgColor = this.story.thumbnailBgColor;
+      this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+        this.story.thumbnailIconUrl
+      );
+      this.progress = 0;
+      this.title = this.story.title;
+
+      if (this.story instanceof CollectionSummary) {
+        this.lessonUrl = '';
+        this.lessonTopic = 'Collections';
+      } else {
+        this.lessonUrl = `/explore/${this.story.id}`;
+        this.lessonTopic = 'Community Lessons';
+      }
+    }
+  }
+
+  getProgress(arg: number): string {
+    let leftCircle = 50 >= arg ? arg : 50;
+    let rightCircle = arg > 50 ? arg - 50 : 0;
+
+    return `linear-gradient(${50 >= arg ? 180 + (arg / 100) * 180 : 270}deg, #00645c ${leftCircle}%, transparent ${leftCircle}%), linear-gradient(0deg, #00645c ${rightCircle}%, lightgray ${rightCircle}%)`;
+  }
+
+  getStorySummaryThumbnailUrl({filename, id}: ThumbnailUrl): string {
+    if (filename) {
```

Comment:
please handle edge cases first

```
if (edge_case1) {
  return result1
}
if (edge_case2) {
  return result2;
}
<general_code>
```
This seriously increases readability, almost all the time

Reply:
Done.